src/services/handlers/mcp-tools.handler.ts

```diff
@@ -22,6 +23,42 @@ export class McpToolsHandler extends McpHandlerBase {
     super(moduleRef, registry, McpToolsHandler.name);
   }
 
+  private buildDefaultContentBlock(result: any) {
+    return [
+      {
+        type: 'text',
+        text: JSON.stringify(result, null, 2),
+      },
+    ];
+  }
+
+  private formatToolResult(result: any, outputSchema?: ZodTypeAny): any {
+    if (result && typeof result === 'object' && Array.isArray(result.content)) {
+      return result;
+    }
+
+    if (outputSchema) {
+      const validation = outputSchema.safeParse(result);
+      if (!validation.success) {
+        return {
+          content: [{
+            type: 'text',
+            text: `Tool result does not match outputSchema: ${validation.error.message}`,
+          }],
+          isError: true,
+        };
```

Comment:
If you throw here the catch block will pick this up and return tne formatted error.

```suggestion
        throw new McpError(
          ErrorCode.InternalError,
          `Tool result does not match outputSchema: ${validation.error.message}`
        );
```